syntax = "proto3";

package fileservice;
option go_package = "gwbench/proto";

service FileService {
  rpc UploadFile(stream FileChunk) returns (UploadResponse);
  rpc DownloadFile(DownloadRequest) returns (stream FileChunk);
}

message FileChunk {
  string filename = 1;
  bytes data = 2;
  int64 total_size = 3;
}

message UploadResponse {
  string message = 1;
  int64 bytes_received = 2;
}

message DownloadRequest {
  string filename = 1;
}